horovod/tensorflow/mpi_ops.cc

```diff
@@ -256,6 +257,8 @@ const common::TensorShape TFTensor::shape() const {
 
 const void* TFTensor::data() const { return (const void*)tensor_.tensor_data().data(); }
 
+const void* TFTensor::tensor() const { return (const void*)&tensor_; }
```

Comment:
Do you need the return type to be a `void*`?  If it's used internally, it might be more convenient to have it as a Tensor pointer.